circuit/src/inputs.rs

```diff
@@ -79,51 +68,53 @@ impl TryFrom<ProofWithPublicInputs<F, C, D>> for PublicCircuitInputs {
 /// All of the private inputs required for the circuit.
 #[derive(Debug)]
 pub struct PrivateCircuitInputs {
-    /// Raw bytes of the nullifier preimage, used to prevent double spends.
-    pub nullifier_preimage: Vec<u8>,
-    /// Raw bytes of the unspendable account preimage.
-    pub unspendable_account_preimage: Vec<u8>,
+    /// Raw bytes of the preimage of the nullifier and the unspendable account
+    pub secret: Vec<u8>,
     /// A sequence of key-value nodes representing the storage proof.
     ///
     /// Each element is a tuple where the items are the left and right splits of a proof node split
     /// in half at the expected childs hash index.
     pub storage_proof: Vec<(Vec<u8>, Vec<u8>)>,
+    pub funding_nonce: u32,
+    pub funding_account: SubstrateAccount,
+    /// The unspendable account hash.
+    pub unspendable_account: UnspendableAccount,
 }
 
 #[cfg(any(test, feature = "testing"))]
 pub mod test_helpers {
-    use crate::exit_account::ExitAccount;
-    use crate::nullifier::{self, Nullifier};
+    use crate::codec::ByteCodec;
+    use crate::substrate_account::SubstrateAccount;
+    use crate::nullifier::Nullifier;
     use crate::storage_proof::test_helpers::{default_proof, ROOT_HASH};
     use crate::unspendable_account::{self, UnspendableAccount};
 
     use super::{CircuitInputs, PrivateCircuitInputs, PublicCircuitInputs};
 
     impl Default for CircuitInputs {
         fn default() -> Self {
-            let nullifier_preimage = hex::decode(nullifier::test_helpers::PREIMAGE).unwrap();
-            let unspendable_account_preimage =
-                hex::decode(unspendable_account::test_helpers::PREIMAGES[0]).unwrap();
+            let secret =
+                hex::decode(unspendable_account::test_helpers::SECRETS[0]).unwrap();
             let root_hash: [u8; 32] = hex::decode(ROOT_HASH).unwrap().try_into().unwrap();
 
-            let nullifier = Nullifier::new(&nullifier_preimage);
-            let unspendable_account = UnspendableAccount::new(&unspendable_account_preimage);
-            let exit_account = ExitAccount::new(&[254u8; 32]).unwrap();
+            let funding_account = SubstrateAccount::new(&[234u8; 32]).unwrap();
+            let nullifier = Nullifier::new(&secret, 0, <[u8; 32]>::try_from(funding_account.to_bytes()).unwrap());
+            let unspendable_account = UnspendableAccount::new(&secret);
+            let exit_account = SubstrateAccount::new(&[254u8; 32]).unwrap();
```

Comment:
I would probably make all of these default values new constants just for clarity